frontend/javascripts/viewer/view/components/setting_input_views.tsx

```diff
@@ -136,209 +131,204 @@ type LogSliderSettingProps = {
-                  defaultChecked={value}
-                  disabled={disabled}
-                  loading={loading}
-                />
-              </FastTooltip>
-              {postSwitchIcon}
-            </div>
-          </FastTooltip>
-          {this.props.children}
-        </Col>
-      </Row>
-    );
-  }
+export function SwitchSetting(props: SwitchSettingProps) {
+  const {
+    label,
+    onChange,
+    value,
+    disabled = false,
+    tooltipText = null,
+    loading = false,
+    labelSpan = null,
+    postSwitchIcon = null,
+    disabledReason,
+    children,
+  } = props;
+
+  const leftSpanValue = labelSpan || SETTING_LEFT_SPAN;
+  const rightSpanValue = labelSpan != null ? ANTD_TOTAL_SPAN - leftSpanValue : SETTING_RIGHT_SPAN;
+  return (
+    <Row className="margin-bottom" align="middle" gutter={ROW_GUTTER}>
+      <Col span={leftSpanValue}>
+        <label className="setting-label">{label}</label>
+      </Col>
+      <Col span={rightSpanValue}>
+        <FastTooltip title={tooltipText} placement="top">
+          {/* This div is necessary for the tooltip to be displayed */}
+          <div
+            style={{
+              display: "inline-flex",
+              justifyContent: "center",
+              alignItems: "center",
+            }}
+          >
+            <FastTooltip title={disabledReason}>
+              <Switch
+                onChange={onChange}
+                checked={value}
+                defaultChecked={value}
+                disabled={disabled}
+                loading={loading}
+              />
+            </FastTooltip>
+            {postSwitchIcon}
+          </div>
+        </FastTooltip>
+        {children}
+      </Col>
+    </Row>
+  );
 }
+
 type NumberInputSettingProps = {
   onChange: (value: number | null) => void;
-  value: number | "";
+  value: number;
   label: string;
   max?: number;
   min?: number;
   step?: number;
 };
-export class NumberInputSetting extends React.PureComponent<NumberInputSettingProps> {
-  static defaultProps = {
-    max: undefined,
-    min: 1,
-    step: 1,
-  };
 
-  render() {
-    const { onChange, value, label, max, min, step } = this.props;
-    return (
-      <Row className="margin-bottom" align="top" gutter={ROW_GUTTER}>
-        <Col span={SETTING_LEFT_SPAN}>
-          <label className="setting-label">{label}</label>
-        </Col>
-        <Col span={SETTING_RIGHT_SPAN}>
-          <InputNumber
-            style={{
-              width: "100%",
-            }}
-            min={min}
-            max={max}
-            onChange={onChange}
-            // @ts-expect-error ts-migrate(2322) FIXME: Type 'number | ""' is not assignable to type 'numb... Remove this comment to see the full error message
-            value={value}
-            step={step}
-            size="small"
-            variant="borderless"
-          />
-        </Col>
-      </Row>
-    );
-  }
+// TODO This component is not used by any other component. Remove it?
```

Comment:
Jup, seems like it. So lets remove it :tada:

Reply:
Done. removed.